### ISSUE TITLE
"Change Signature" doesn't add lifetime declaration to the function

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#7082]
## Environment

* **IntelliJ Rust plugin version:** 0.3.146.3786-211-nightly
* **Rust toolchain version:** 1.52.0-nightly (b3e19a221 2021-03-12) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.1 (CL-211.6693.114)
* **Operating system:** macOS 10.15.6
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

If I add a new parameter with a lifetime, "change signature" leaves the lifetime undeclared.

## Steps to reproduce

```rust
fn foo() {}
```

Call change signature and add a new parameter `p0` with `&'a i32`

```rust
fn foo(p0: &'a i32) {}
```

https://user-images.githubusercontent.com/2500320/114225237-c4198a00-997a-11eb-99d1-0170e056ab30.mov

**Comment by lancelote** [comment]
Actually, I think there are more relevant problems with lifetimes, e.g.

- Lifetime declarations are not removed once unused (for example, I remove the parameter using a lifetime)
- Declarations are not updated after renaming a lifetime

P.S. Sorry for the spam of the "change signature" tickets. The feature is awesome 👍

**Comment by Kobzol** [comment]
Right. So I actually considered changing lifetimes and (especially) generic parameters to be out of scope, because the user can write pretty much arbitrary things into the parameter/return types and it would be difficult to distinguish and consolidate generic types from unknown types.

But I suppose that we could do better at least w.r.t. lifetimes.

I wonder if we should do this automatically or just let the user edit the lifetime/generic parameters block directly in the refactoring dialog?

**Comment by lancelote** [comment]
As a lazy person, I would prefer it to be automatic :3 Though it makes it harder to implement indeed :(

**Comment by Kobzol** [comment]
I guess that removing unused lifetimes should be relatively easy to handle.

Detecting lifetime renames is quite harder, because it's difficult to recognize if a lifetime was introduced, renamed or moved to a new location. The same thing is also true for new lifetimes.
